internal/controllers/import_controller.go

```diff
@@ -472,69 +472,35 @@ func (r *CAPIImportReconciler) createImportManifest(ctx context.Context, remoteC
 }
 
 func (r *CAPIImportReconciler) createRawManifest(ctx context.Context, remoteClient client.Client, bytes []byte) error {
-	bytes, err := yamlDecoder.ToJSON(bytes)
+	items, err := utilyaml.ToUnstructured(bytes)
 	if err != nil {
-		return err
-	}
-
-	check := map[string]interface{}{}
-	if err := json.Unmarshal(bytes, &check); err != nil {
 		return fmt.Errorf("error unmarshalling bytes or empty object passed: %w", err)
 	}
 
-	// return if object is empty
-	if len(check) == 0 {
-		return nil
-	}
-
-	obj, _, err := unstructured.UnstructuredJSONScheme.Decode(bytes, nil, nil)
-	if err != nil {
-		return err
-	}
-
-	switch obj := obj.(type) {
-	case *unstructured.Unstructured:
-		if err := r.createObject(ctx, remoteClient, obj); err != nil {
+	for _, obj := range items {
+		if err := r.createObject(ctx, remoteClient, obj.DeepCopy()); err != nil {
 			return err
 		}
-
-		return nil
-	case *unstructured.UnstructuredList:
-		for i := range obj.Items {
-			if err := r.createObject(ctx, remoteClient, &obj.Items[i]); err != nil {
-				return err
-			}
-		}
-
-		return nil
-	default:
-		return fmt.Errorf("unknown object type: %T", obj)
 	}
+
+	return nil
 }
 
 func (r *CAPIImportReconciler) createObject(ctx context.Context, c client.Client, obj client.Object) error {
 	log := log.FromContext(ctx)
 	gvk := obj.GetObjectKind().GroupVersionKind()
 
-	u := &unstructured.Unstructured{}
-	u.SetGroupVersionKind(gvk)
-	err := c.Get(ctx, client.ObjectKey{
-		Namespace: obj.GetNamespace(),
-		Name:      obj.GetName(),
-	}, u)
+	err := c.Create(ctx, obj)
+	if client.IgnoreAlreadyExists(err) != nil {
```

Comment:
Whats the difference between this check and the `apierrors.IsAlreadyExists` below?

Reply:
The first one filters out the AlreadyExists error, and returns all other error types or nil. The other one specifically checks for the error being of type AlreadyExist. I hoped this will be more readable, but may be it isn’t. Can change to err != nil for the second one, it is equivalent.

Reply:
we could move maybe this:

`if apierrors.IsAlreadyExists(err) {}` up to check for api errors and later check for err != nil since it will be not api errors anymore?

Reply:
I’ve updated the implementation. Though this change brings no functional difference, just the code style.